server/src/document/document.controller.ts

```diff
@@ -65,4 +76,80 @@ export class DocumentController {
       response.status(400).send({ "error": 'You can only upload files to dcp_communityboarddisposition at this time' });
     }
   }
+
+  // "path" refers to the "relative server path", the path
+  // to the file itself on the sharepoint host. It has the format
+  //
+  //   /sites/<site>/<entity_name>/<folder_name>/filename.pdf
+  //
+  // For example,
+  //
+  //   /sites/dcpuat2/dcp_package/2021M0268_DraftEAS_1_996699F37323EB11A813001DD8309FA8/EAS%20Full%20Form.pdf
+  //
+  // Note that <folder_name> has this format:
+  //
+  //   <package_name>_<PACKAGEID>
+  //
+  // where package_name is the dcp_name property on the package
+  // (spaces replaced with underscores)
+  // and PACKAGEID is the package dcp_packageid property, stripped
+  // of hyphens.
+  //
+  // We have assurance of this. See
+  // https://dcp-paperless.visualstudio.com/dcp-paperless-dynamics/_workitems/edit/13366
+  @Get('/*')
+  async read(@Param() path, @Res() res) {
+    const pathSegment = path[0];
+    const packageFolder = pathSegment.split('/')[3];
+    const packageFolderSegments = packageFolder.split('_');
+    const strippedPackageId = packageFolderSegments[packageFolderSegments.length - 1];
+
+    // we need to re-insert hyphens to recreate the actual packageId
+    // TODO: Consider asking to preseve hyphens in Package ID
+    const packageId = [
+      strippedPackageId.slice(0, 8),
+      '-',
+      strippedPackageId.slice(8, 12),
+      '-',
+      strippedPackageId.slice(12,16),
+      '-',
+      strippedPackageId.slice(16,20),
+      '-',
+      strippedPackageId.slice(20),
+    ].join('');
```

Comment:
This kind of stuff should at least go in a different function if not the service?

Reply:
true!! can do

Reply:
Make a function outside the class called "formatPackageToHyphenated" or something. Use your comments to make a function with that sort of name.

Reply:
What do we know about these kinds of GUIDs? Do they follow a standard? Maybe there's the equivalent of something like "dasherize"

Reply:
True they might follow a standard. It's been pretty consistent that they have the hyphens in those specific indexes. 
According to Microsoft docs:
https://docs.microsoft.com/en-us/dynamics-nav/guid-data-type

```
The GUID is a 16-byte binary data type that can be logically grouped into the following subgroups:

4byte-2byte-2byte-2byte-6byte.

The standard textual representation is {12345678-1234-1234-1234-1234567890AB}.
```

So yeah maybe there is a util already.. i can check